centreon/www/front_src/src/Resources/useDateTimePickerAdapter.ts

```diff
@@ -196,6 +214,45 @@ const useDateTimePickerAdapter = (): UseDateTimePickerAdapterProps => {
       );
     };
 
+    public getWeekArray = (date: dayjs.Dayjs): Array<Array<dayjs.Dayjs>> => {
+      if (date.isUTC()) {
+        const start = date.utc().startOf('month').startOf('week');
+        const end = date.utc().endOf('month').endOf('week');
+
+        let count = 0;
+        let current = start;
+        const nestedWeeks: Array<Array<dayjs.Dayjs>> = [];
+
+        while (current.isBefore(end)) {
+          const weekNumber = Math.floor(count / 7);
+          nestedWeeks[weekNumber] = nestedWeeks[weekNumber] || [];
+          nestedWeeks[weekNumber].push(current);
+
+          current = current.add(1, 'day') as dayjs.Dayjs;
+          count += 1;
+        }
+
+        return nestedWeeks;
+      }
+      const start = date.startOf('month').startOf('week');
+      const end = date.endOf('month').endOf('week');
+
+      let count = 0;
+      let current = start;
+      const nestedWeeks: Array<Array<dayjs.Dayjs>> = [];
+
+      while (current.isBefore(end)) {
+        const weekNumber = Math.floor(count / 7);
+        nestedWeeks[weekNumber] = nestedWeeks[weekNumber] || [];
+        nestedWeeks[weekNumber].push(current);
+
+        current = current.tz(timezone, true).add(1, 'day') as dayjs.Dayjs;
+        count += 1;
+      }
```

Comment:
- Avoid mutation (use a functional approach, using rambda for instance)
- Try to reuse a function when the code is similar to improve reusability